packages/yew/src/io_coop.rs

```diff
@@ -0,0 +1,27 @@
+//! module that provides io compatibility over browser tasks and other async io tasks (e.g.: tokio)
```

Comment:
Can we have a similar module that provides `Rc` and `RefCell` in browser and `Arc` and `RwLock` on server? That should solve the `!Send` problem

Reply:
> Can we have a similar module that provides Rc and RefCell in browser and Arc and RwLock on server?

Due to Yew's Virtual Dom design, it's nearly impossible to separate `web_sys::*` usage from virtual dom.

In addition, it means that to support SSR, all components when used on the server side also needs to be `Send` which means that it essentially imposes a requirement for developer to flood their application with feature flags to prevent any `web_sys` in their component state.
(I am not willing to add an endless number of feature flags for my application and I can live without SSR if that is the case.)

Reply:
That's a good point. I'd hate to have to put `#[cfg]`s everywhere.

There was a previous attempt at refactoring out rendering back end so that web_sys and vdom weren't so interconnected: https://github.com/yewstack/yew/pull/1719. I don't know if it solves this problem but could be worth looking at

Reply:
I think the bigger issue here is that `!Send` types cannot be carried over an await point.

As long as we have a requirement to keep component state while rendering (Scope is needed to render) and render is asynchronous, then `render()` needs to be `!Send`.

Reply:
Oh well... I was hoping maybe there could be a way to get Send future. It's fine as-is. SSR being implemented is a big step forward.